Jenkinsfile.deploy

```
@@ -160,7 +163,12 @@ pipeline {
         stage('Deploy artifacts') {
             steps {
                 script {
-                    if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
+                    if (params.MAVEN_DEPLOY_REPOSITORY && params.MAVEN_REPO_CREDS_ID) {
+                        // Deploy to specific repository with credentials
+                        runMavenDeployLocally(getMavenCommand())
+                        maven.uploadLocalArtifacts(params.MAVEN_REPO_CREDS_ID, getLocalDeploymentFolder(), getMavenRepoZipUrl())
+                    } else if(!isRelease() || params.MAVEN_DEPLOY_REPOSITORY){
+                        // Normal deploy
                         runMavenDeploy(getMavenCommand())
```

I guess for clarity in the code, we should put the `directory` parameter as mandatory ...

```suggestion
                        runMavenDeployLocally(getMavenCommand(getRepoName()))
```